File: Upcoder/Lập trình cơ bản/5.Mảng/Mang ZicZac/MANGCS4.cs
using System;
using System.Linq;
namespace Ziczac
{
    public class Program
    {
        static void Main(string[]args)
        {
            var dims = Console.ReadLine().Split().Select(int.Parse).ToArray();
            int n = dims[0], m = dims[1];
            
            int[,] a = new int[n, m];
            
            for (int i = 0; i < n; i++)
            {
                var row = Console.ReadLine().Split().Select(int.Parse).ToArray();
                for (int j = 0; j < m; j++)
                {
                    a[i, j] = row[j];
                }
            }

            int[] temp = new int[n * m];
            int index = 0;
            
            for (int i = 0; i < n; i++)
            {
                for (int j = 0; j < m; j++)
                {
                    temp[index++] = a[i, j];
                }
            }

            Array.Sort(temp);

            index = 0;
            for (int i = 0; i < n; i++)
            {
                for (int j = 0; j < m; j++)
                {
                    a[i, j] = temp[index++];
                }
            }

            for (int i = 0; i < n; i++)
            {
                if (i % 2 == 0)
                {
                    for (int j = 0; j < m; j++)
                    {
                        Console.Write(a[i, j] + " ");
                    }
                }
                else
                {
                    for (int j = m - 1; j >= 0; j--)
                    {
                        Console.Write(a[i, j] + " ");
                    }
                }
                Console.WriteLine();
            }
        }
    }
}

File: Upcoder/Lập trình cơ bản/5.Mảng/Mang ZicZac/MANGCS4.py
n, m = map(int, input().split())

a = []
for i in range(n):
    row = list(map(int, input().split()))
    a.append(row)

temp = []
for i in range(n):
    for j in range(m):
        temp.append(a[i][j])

temp.sort()

index = 0
for i in range(n):
    for j in range(m):
        a[i][j] = temp[index]
        index += 1

for i in range(n):
    if i % 2 == 0:
        print(*a[i])
    else:
        print(*a[i][::-1])
